### PR TITLE
feat(toolchain): Add support for Python 3.13.1.

### PR DESCRIPTION
**Pull request by okin**
Adds 3.13.1 Python toolchain.

Also updates 3.13 to map to 3.13.1.

Tests have been slightly altered to make spotting the override easier.

### CODE REVIEW

**Review by aignas — CHANGES_REQUESTED** [review]
Thanks for the PR!

Please add other versions for the toolchains from the latest release and use the [20241206](https://github.com/indygreg/python-build-standalone/releases/tag/20241206) release.

That means:
* CPython 3.9.20 -> 3.9.21
* CPython 3.10.15 -> 3.10.16
* CPython 3.11.10 -> 3.11.11
* CPython 3.12.7 -> 3.12.8
* CPython 3.13.0 -> 3.13.1

Also please add a note to the CHANGELOG.md just like it was done for previous releases.

**Comment by rickeylev** [comment]
Thanks for the PR!

I pushed an update for the changelog for the 3.13.1 part and synced this to head.

File: CHANGELOG.md [hunk 1]
```diff
@@ -83,7 +84,14 @@ Unreleased changes template.
   {obj}`experimental_index_url` usage or the regular `pip.parse` usage.
   To select the free-threaded interpreter in the repo phase, please use
   the documented [env](/environment-variables.html) variables.
-  Fixes [#2386](https://github.com/bazelbuild/rules_python/issues/2386).
+  Fixes [#2386](https://github.com/bazelbuild/rules_python/issues/2386).* (toolchains) Python 3.13.1 added
+* (toolchains) Use the latest astrahl-sh toolchain release [20241206] for Python versions:
+    * 3.9.21
+    * 3.10.16
+    * 3.11.11
+    * 3.12.8
```
Comment (aignas) [inline CHANGELOG.md:92]:
```suggestion
  Fixes [#2386](https://github.com/bazelbuild/rules_python/issues/2386).
* (toolchains) Use the latest astrahl-sh toolchain release [20241206] for Python versions:
    * 3.9.21
    * 3.10.16
    * 3.11.11
    * 3.12.8
    * 3.13.1
```

Technically speaking these are new versions, so we can just lump the changelog entries.

**Review by aignas — APPROVED** [review]

**Comment by aignas** [comment]
Thank you!

**Pull request merged**